db/sqlinterfaces.c

```diff
@@ -8530,181 +8462,34 @@ static int execute_sql_query_offload_inner_loop(struct sqlclntstate *clnt,
-    const char *rest_of_sql = NULL;
-    struct sql_thread *thd = poolthd->sqlthd;
-    sqlite3 *sqldb = poolthd->sqldb;
-    char *sql = clnt->sql;
-    char *sql_str = NULL;
-    struct sql_thread *thdcmp = pthread_getspecific(query_info_key);
+    int rc = 0, ret = 0;
     char *errstr = NULL;
-    struct schema *parameters_to_bind = NULL;
-    int have_our_own_error = 0;
-    int sqlcache_hint = 0;
-    char cache_hint[128];
-    stmt_hash_entry_type *stmt_entry = NULL;
-    char *err = NULL;
-    int i;
-    int sql_cache_hint_found = 0;
-    int hint_len;
-
-    /* asserts */
-    if (thdcmp != thd) {
-        logmsg(LOGMSG_ERROR, "%s:Wrong sql_thread for the wrong sql thread!\n",
-                __func__);
-        return SQLITE_INTERNAL;
-    }
-    if (!thd || !sqldb) {
-        logmsg(LOGMSG_ERROR, "%s: no sql_thread or sqlite3 structure\n", __func__);
+    struct sql_thread *thd = poolthd->sqlthd;
+    if (!thd) {
+        logmsg(LOGMSG_ERROR, "%s: no sql_thread\n", __func__);
         return SQLITE_INTERNAL;
     }
-
-    if (clnt->is_newsql) {
-        ATOMIC_ADD(gbl_nnewsql, 1);
-    } else {
-        ATOMIC_ADD(gbl_nsql, 1);
-    }
-    thd->startms = time_epochms();
-    thd->stime = time_epoch();
-    thd->nmove = thd->nfind = thd->nwrite = 0;
-
-    if (clnt->tag) {
-        parameters_to_bind = new_dynamic_schema(clnt->tag, strlen(clnt->tag),
-                                                gbl_dump_sql_dispatched);
-        if (parameters_to_bind == NULL) {
-            logmsg(LOGMSG_ERROR, "%s:%d invalid parametrized sql tag: %s\n", __FILE__,
-                   __LINE__, clnt->tag);
-            return SQLITE_INTERNAL;
-        }
-    }
-
-    reqlog_new_sql_request(poolthd->logger, sql);
+    reqlog_new_sql_request(poolthd->logger, clnt->sql);
     log_queue_time(poolthd->logger, clnt);
-
-    rc = sql_set_transaction_mode(sqldb, clnt, clnt->dbtran.mode);
-    if (rc) {
-        ret = FSQL_PREPARE;
-        err = sqlite3_mprintf("Failed to set transaction mode.");
-        have_our_own_error = 1;
-        goto done_here;
-    }
-
-    if (clnt->using_case_insensitive_like)
-        toggle_case_sensitive_like(sqldb, 1);
-
-    /* reset error */
     bzero(&clnt->fail_reason, sizeof(clnt->fail_reason));
     bzero(&clnt->osql.xerr, sizeof(clnt->osql.xerr));
-
-    if (clnt->rawnodestats)
-        clnt->rawnodestats->sql_queries++;
-
-    thrman_wheref(poolthd->thr_self, "%s", clnt->sql);
-
+    struct sql_state rec = {0};
+    rec.sql = clnt->sql;
+    rc = get_prepared_bound_stmt(poolthd, clnt, &rec, &clnt->osql.xerr);
+    if (rc) {
+        return rc;
+    }
+    sqlite3 *sqldb = poolthd->sqldb;
+    sqlite3_stmt *stmt = rec.stmt;
+    thrman_wheref(poolthd->thr_self, "%s", rec.sql);
     user_request_begin(REQUEST_TYPE_QTRAP, FLAG_REQUEST_TRACK_EVERYTHING);
-
     if (gbl_dump_sql_dispatched)
         logmsg(LOGMSG_USER, "BLOCKSQL mode=%d [%s]\n", clnt->dbtran.mode,
                 clnt->sql);
-
```

Comment:
thank you following up on this, I have not get to it after execute_sql_query refactoring!

Reply:
Yes this was a fun one, as I am sure it was for you before!